base64ct/src/lib.rs

```diff
@@ -353,23 +353,27 @@ fn encode<'a>(
 
     let dst = &mut dst[..elen];
 
+    let mut src_chunks = src.chunks_exact(3);
+    let mut dst_chunks = dst.chunks_exact_mut(4);
+
+    for (s, d) in (&mut src_chunks).zip(&mut dst_chunks) {
+        encode_3bytes(s, d, hi_bytes);
+    }
+
+    let src_rem = src_chunks.remainder();
+
     if padded {
-        for (s, d) in src.chunks(3).zip(dst.chunks_mut(4)) {
-            if s.len() == 3 {
-                encode_3bytes(s, d, hi_bytes);
-            } else {
-                encode_3bytes_padded(s, d, hi_bytes);
-            }
+        if let Some(dst_rem) = dst_chunks.next() {
+            let mut tmp = [0u8; 3];
+            tmp[..src_rem.len()].copy_from_slice(&src_rem);
+            encode_3bytes(&tmp, dst_rem, hi_bytes);
+
+            let flag = src_rem.len() == 1;
+            let mask = (flag as u8).wrapping_sub(1);
+            dst_rem[2] = (dst_rem[2] & mask) | (PAD & !mask);
```

Comment:
Also note that I have replaced the explicit branch with this bit-twiddling. I wonder if there is a better way to generate mask from `bool`.